File: serverTCC/serverTCC/Models/Investimento.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace serverTCC.Models
{
    public class Investimento
    {
        public int Id { get; set; }

        public Usuario Usuario { get; set; }
        public string UsuarioId { get; set; }

        public TipoInvestimento TipoInvestimento { get; set; }
        public int TipoInvestimentoId { get; set; }

        public Moeda Moeda { get; set; }
        public int MoedaId { get; set; }

        public string Descricao { get; set; }
        public decimal ValorAtual { get; set; }
        public decimal ValorInvestido { get; set; }
        public int QtdTempo { get; set; }
        public DateTime DataInicio { get; set; }

        public EscalaTempo EscalaTempo { get; set; }

        [JsonIgnore]
        public DateTime UltimaAtualizacao { get; set; }
        [JsonIgnore]
        public List<ValoresInseridos> ValoresInseridos { get; set; }
    }

    public class ValoresInseridos
    {
        public int Id { get; set; }
        public int InvestimentoId { get; set; }
        public Investimento Investimento { get; set; }
        public DateTime Data { get; set; }
        public decimal Valor { get; set; }
    }
}

File: client/www/js/models/investimento.js
function Investimento(data) {
    "use strict";
    if (!data) data = {};
    
    this.id = data.id;
    if (data.usuario)
        this.usuario = new Usuario(data.usuario);
    this.usuarioId = data.usuarioId;
    if (data.tipoInvestimento)
        this.tipoInvestimento = new TipoInvestimento(data.tipoInvestimento);
    this.tipoInvestimentoId = data.tipoInvestimentoId;
    if (data.moeda)
        this.moeda = new Moeda(data.moeda);
    this.moedaId = data.moedaId;
    this.descricao = data.descricao;
    this.valorAtual = data.valorAtual;
    this.valorInvestido = data.valorInvestido;
    this.qtdTempo = data.qtdTempo;
    this.dataInicio = data.dataInicio;
    this.escalaTempo = data.escalaTempo;
}

localEntities.register(Investimento, {
    id: {
        type: 'number',
        pk: true
    },
    usuario: {
        type: 'Usuario',
        relationKey: 'usuarioId'
    },
    usuarioId: {
        type: 'number',
        fk: true,
    },
    tipoInvestimento: {
        type: 'TipoInvestimento',
        relationKey: 'tipoInvestimentoId'
    },
    tipoInvestimentoId: {
        type: 'number',
        fk: true
    },
    moeda: {
        type: 'Moeda',
        relationKey: 'moedaId'
    },
    moedaId: {
        type: 'number',
        fk: true
    },
    descricao: {
        type: 'string'
    },
    valorAtual: {
        type: 'number'
    },
    valorInvestido: {
        type: 'number'
    },
    qtdTempo: {
        type: 'number'
    },
    dataInicio: {
        type: 'Date'
    },
    escalaTempo: {
        type: 'number'
    },
});